src/lib.rs

```diff
@@ -165,6 +181,36 @@ pub struct TextInputSettings {
 #[derive(Component, Default, Reflect)]
 pub struct TextInputValue(pub String);
 
+/// A component containing the placeholder text that is displayed when the text input is empty.
+#[derive(Component, Default, Reflect)]
+pub struct TextInputPlaceholder {
+    /// The placeholder text.
+    pub value: String,
+    /// The style to use when rendering the placeholder text.
+    pub text_style: Option<TextStyle>,
+}
+
+impl TextInputPlaceholder {
+    /// Returns the style to use when rendering the placeholder text.
+    /// Uses the own style if it exists, otherwise uses the input style with half opacity.
```

Comment:
and here might need to be updated
```suggestion
    /// Uses the own style if it exists, otherwise uses the input style with quarter opacity.
```